integrations/jta/jdbc/src/main/java/io/helidon/integrations/jta/jdbc/LocalXAResource.java

```diff
@@ -0,0 +1,1031 @@
+        }
+
+        switch (flags) {
+        case TMENDRSCAN:
+        case TMNOFLAGS:
+        case TMSTARTRSCAN:
+            break;
+        default:
+            // Bad flags.
+            throw (XAException) new XAException(XAER_INVAL)
+                .initCause(new IllegalArgumentException("flags: " + flagsToString(flags)));
+        }
+
+        if (LOGGER.isLoggable(Level.FINER)) {
+            LOGGER.exiting(this.getClass().getName(), "recover", EMPTY_XID_ARRAY);
+        }
+        return EMPTY_XID_ARRAY;
+    }
+
+    @Override // XAResource
+    public boolean isSameRM(XAResource xaResource) throws XAException {
+        if (LOGGER.isLoggable(Level.FINER)) {
+            LOGGER.entering(this.getClass().getName(), "isSameRM", xaResource);
+            LOGGER.exiting(this.getClass().getName(), "isSameRM", this == xaResource);
+        }
+        return this == xaResource;
+    }
+
+    @Override // XAResource
+    public int getTransactionTimeout() {
+        if (LOGGER.isLoggable(Level.FINER)) {
+            LOGGER.entering(this.getClass().getName(), "getTransactionTimeout");
+            LOGGER.exiting(this.getClass().getName(), "getTransactionTimeout", 0);
+        }
+        return 0;
+    }
+
+    @Override // XAResource
+    public boolean setTransactionTimeout(int transactionTimeoutInSeconds) {
+        // (Interesting note: this is the first method that is called by the TransactionManager, i.e. before #start(Xid,
+        // int).)
+        if (LOGGER.isLoggable(Level.FINER)) {
+            LOGGER.entering(this.getClass().getName(), "setTransactionTimeout", transactionTimeoutInSeconds);
+            LOGGER.exiting(this.getClass().getName(), "setTransactionTimeout", false);
+        }
+        return false;
+    }
+
+    private Association computeAssociation(XARoutine xaRoutine,
+                                           Xid xid,
+                                           BiFunction<? super Xid, ? super Association, ? extends Association> f)
+        throws XAException {
+        try {
+            return ASSOCIATIONS.compute(xid, f);
+        } catch (RuntimeException e) {
+            throw this.convert(xaRoutine, e);
+        }
+    }
+
+    private Association computeAssociation(XARoutine xaRoutine,
+                                           Xid xid,
+                                           EnumSet<Association.BranchState> legalBranchStates,
+                                           UnaryOperator<Association> f,
+                                           boolean removeAssociationOnError)
+        throws XAException {
+        try {
+            return ASSOCIATIONS.compute(xid, (x, a) -> remap(x, a, legalBranchStates, f));
+        } catch (RuntimeException e) {
+            if (removeAssociationOnError) {
+                ASSOCIATIONS.remove(xid);
+            }
+            throw this.convert(xaRoutine, e);
+        }
+    }
+
+    private XAException convert(XARoutine xaRoutine, Throwable e) {
```

Comment:
After reading many bug reports from different decades it's clear that no one, not even the specification authors, has made the error requirements clear to implementors, so this method plus the `ExceptionConverter` interface should allow an escape hatch for us if we end up getting the default behavior wrong.